pkg/executor/batch_point_get.go

```diff
@@ -235,6 +236,10 @@ func (e *BatchPointGetExec) initialize(ctx context.Context) error {
 	var indexKeys []kv.Key
 	var err error
 	batchGetter := e.batchGetter
+	if e.Ctx().GetSessionVars().MaxExecutionTime > 0 {
+		// If MaxExecutionTime is set, we need to set the context deadline for the batch get.
+		ctx, _ = context.WithTimeout(ctx, time.Duration(e.Ctx().GetSessionVars().MaxExecutionTime)*time.Millisecond)
```

Comment:
It is safer to to use
```
ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
defer cancel()
```
to follow the best practise and avoid unexpected resource usage risks.

Reply:
yes, will do to avoid context leakage